aloha-core/src/main/scala/com/eharmony/aloha/io/sources/modelSources.scala

```diff
@@ -0,0 +1,114 @@
+package com.eharmony.aloha.io.sources
+
+import java.net.URLEncoder
+import java.security.MessageDigest
+
+import com.eharmony.aloha.io.vfs.VfsType.VfsType
+import com.eharmony.aloha.io.vfs.{File, Vfs, VfsType}
+import org.apache.commons.codec.binary.Base64
+import spray.json._
+
+import scala.util.Try
+
+
+sealed trait ModelSource {
+  def localVfs: File
+  def shouldDelete: Boolean
+}
+
+object ModelSource {
+
+  /**
+   * Allows for the following forms.
+   * {{{
+   * { "model":    "[Base 64 encoded byte array here]" }
+   * { "modelUrl": "some://apache.vfs2/url/here" }
+   * { "modelUrl": "some://apache.vfs1/url/here", "via": "vfs1" }
+   * { "modelUrl": "some://apache.vfs2/url/here", "via": "vfs2" }
+   * { "modelUrl": "/absolute/path/here",         "via": "file" }
+   * }}}
+   */
+  implicit object jsonFormat extends RootJsonFormat[ModelSource] with DefaultJsonProtocol {
```

Comment:
I have remember reading (and I don't remember where) that it is usually good style to add return types to functions for readability.  What do you think?